include/fdreadoutlibs/wib2/WIB2FrameProcessor.hpp

```diff
@@ -518,49 +583,63 @@ class WIB2FrameProcessor : public readoutlibs::TaskRawDataProcessorModel<types::
-            timestamp + clocksPerTPCTick * (int64_t(hit_end[i]) - hit_tover[i]);       // NOLINT(build/unsigned)
-          uint64_t tp_t_end = timestamp + clocksPerTPCTick * int64_t(hit_end[i]);      // NOLINT(build/unsigned)
-
-          // May be needed for TPSet:
-          // uint64_t tspan = clocksPerTPCTick * hit_tover[i]; // is/will be this needed?
-          //
-
-          // For quick n' dirty debugging: print out time/channel of hits.
-          // Can then make a text file suitable for numpy plotting with, eg:
-          //
-          // sed -n -e 's/.*Hit: \(.*\) \(.*\).*/\1 \2/p' log.txt  > hits.txt
-          //
-          //TLOG_DEBUG(0) << "Hit: " << tp_t_begin << " " << offline_channel;
-
-          triggeralgs::TriggerPrimitive trigprim;
-          trigprim.time_start = tp_t_begin;
-          trigprim.time_peak = (tp_t_begin + tp_t_end) / 2;
-          trigprim.time_over_threshold = hit_tover[i] * clocksPerTPCTick;
-          trigprim.channel = offline_channel;
-          trigprim.adc_integral = hit_charge[i];
-          trigprim.adc_peak = hit_charge[i] / 20;
-          trigprim.detid =
-            m_link; // TODO: convert crate/slot/link to SourceID Roland Sipos rsipos@cern.ch July-22-2021
-          trigprim.type = triggeralgs::TriggerPrimitive::Type::kTPC;
-          trigprim.algorithm = triggeralgs::TriggerPrimitive::Algorithm::kTPCDefault;
-          trigprim.version = 1;
-
-          if (!m_tphandler->add_tp(trigprim, timestamp)) {
-            m_tps_dropped++;
-          }
+          if (m_channel_mask_set.find(offline_channel) == m_channel_mask_set.end()) {   
 
-          m_new_tps++;
-          ++nhits;
+            uint64_t tp_t_begin =                                                        // NOLINT(build/unsigned)
+              timestamp + clocksPerTPCTick * (int64_t(hit_end[i]) - int64_t(hit_tover[i]));       // NOLINT(build/unsigned)
+            uint64_t tp_t_end = timestamp + clocksPerTPCTick * int64_t(hit_end[i]);      // NOLINT(build/unsigned)
+  
+            // May be needed for TPSet:
+            // uint64_t tspan = clocksPerTPCTick * hit_tover[i]; // is/will be this needed?
+            //
+  
+            // For quick n' dirty debugging: print out time/channel of hits.
+            // Can then make a text file suitable for numpy plotting with, eg:
+            //
+            // sed -n -e 's/.*Hit: \(.*\) \(.*\).*/\1 \2/p' log.txt  > hits.txt
+            //
+            //TLOG() << "Hit: " << tp_t_begin << " " << offline_channel;
+  
+            triggeralgs::TriggerPrimitive trigprim;
+            trigprim.time_start = tp_t_begin;
+            trigprim.time_peak = (tp_t_begin + tp_t_end) / 2;
+            trigprim.time_over_threshold = int64_t(hit_tover[i]) * clocksPerTPCTick;
+            trigprim.channel = offline_channel;
+            trigprim.adc_integral = hit_charge[i];
+            trigprim.adc_peak = hit_charge[i] / 20;
```

Comment:
Why do we divide by 20 here? Didn't we do a division in the AVX2 code?
Also, why are adc_integral and adc_peak coming from the same variable, surely one is not just 20* the other every time?

Reply:
OK, makes sense.